### ISSUE TITLE
bug in compute_lm_loglikeli

### ISSUE THREAD

**Issue by wangzizhao** [Linear95/SPAG#13]
If I understand correctly, in https://github.com/Linear95/SPAG/blob/main/trainers.py#L34-L36, the `ignore_mask` should be applied to `loss` to mask out ignored idx, but it's not applied. 
```
ignore_mask = labels != IGNORE_INDEX
avg_loss = loss.sum(dim=-1) / ignore_mask.sum(dim=-1)
```

Do I miss something? Thanks for the clarification in advance!

**Comment by Linear95** [comment]
Refer to the PyTorch document of [CrossEntropyLoss](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html), `loss` is automatically calculated with labels = `IGNORE_INDEX` ignored. So in tensor `loss`, positions with labels = `IGNORE_INDEX` are already zero.

**Comment by wangzizhao** [comment]
Never noticed this specificial argument before. Thanks for the clarification!
